### PR TITLE
First attempt at setting up docker-compose stack.

### PR DESCRIPTION
**Pull request by zelnaut**
* Added docker-compose stack with DB, API, Web and PGAdmin.
* Configure PGAdmin to connect to our DB
* Allow VSCode to debug containers
* Add hot-reloading to API/Web containers
* Use environment variables for config

This PR kinda spiraled out of control :shrug: 

Main thing to check is that you can clone the repo, and run `docker-compose up` and have all the projects running.

Will need to add some docs as well.

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -63,12 +78,17 @@ func Register(c *gin.Context) {
...
+	if err != nil {
+		log.Fatal("Error loading .env file")
+	}
+
 	CreateTables(context.Background())
 
 	// Create gin server
 	r := gin.Default()
 	r.Use(cors.New(cors.Config{
-		AllowOrigins:     []string{"http://localhost:3000", "http://localhost:4000"},
```
Comment (QuickFinesse) [inline api/main.go:71]: I had to add this back to `AllowOrigins:     []string{"http://localhost:3000", "http://localhost:4000"},` to get it to work in Windows

File: api/main.go [hunk 2]
```diff
@@ -63,12 +78,17 @@ func Register(c *gin.Context) {
...
+		AllowOrigins:     []string{"*"},
```
Comment (QuickFinesse) [inline api/main.go:91]: I had to add this back to the code below to get it to work in Windows
```suggestion
		AllowOrigins:     []string{"http://localhost:3000", "http://localhost:4000"},
```

**Review by QuickFinesse — CHANGES_REQUESTED** [review]
Works fucking great! Only thing is  the cors error I updated below. You'll have to double check that works on your mac or linux whatever you're using

File: api/main.go (hunk 2 above)
Comment (zelnaut) [reply api/main.go:91]: Just tried it out again locally, for some reason it works fine through Firefox, but not Chrome :shrug:

**Review by QuickFinesse — APPROVED** [review]
👍

**Pull request merged**